From ac029a0ee4b58b77b3b517f1d52dad99633ced6f Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Wed, 17 May 2017 14:03:11 -0700

This involves adding bootcount in environment, as well as modifying
the default bootconfiguration to run ${mender_setup} first.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 configs/ni_sulfur_rev3_defconfig |  2 +-
 include/configs/ni_sulfur_rev3.h | 20 +++++++++-----------
 2 files changed, 10 insertions(+), 12 deletions(-)

diff --git a/configs/ni_sulfur_rev3_defconfig b/configs/ni_sulfur_rev3_defconfig
index 2a6a519..42bb18e 100644
--- a/configs/ni_sulfur_rev3_defconfig
+++ b/configs/ni_sulfur_rev3_defconfig
@@ -3,7 +3,7 @@ CONFIG_SYS_VENDOR="ni"
 CONFIG_SYS_CONFIG_NAME="ni_sulfur_rev3"
 CONFIG_ARCH_ZYNQ=y
 CONFIG_DEFAULT_DEVICE_TREE="zynq-ni-sulfur-rev3"
-CONFIG_SYS_EXTRA_OPTIONS="ENV_IS_IN_MMC"
+CONFIG_SYS_EXTRA_OPTIONS="BOOTCOUNT_LIMIT,BOOTCOUNT_ENV,ENV_IS_IN_MMC"
 CONFIG_FIT=y
 # CONFIG_DISPLAY_CPUINFO is not set
 CONFIG_SPL=y
diff --git a/include/configs/ni_sulfur_rev3.h b/include/configs/ni_sulfur_rev3.h
index 46715bc..e120dbc 100644
--- a/include/configs/ni_sulfur_rev3.h
+++ b/include/configs/ni_sulfur_rev3.h
@@ -14,27 +14,25 @@
 #include "zynq-common.h"
 #undef CONFIG_ENV_IS_NOWHERE
 
-#define CONFIG_SYS_MMC_ENV_DEV 0
-#define CONFIG_ENV_SECT_SIZE		CONFIG_ENV_SIZE
-#define CONFIG_ENV_OFFSET		0xE0000
-
 #undef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"fit_image=fit.itb\0"		\
+	"variant=sulfur-rev3\0"	\
+	"fit_image=boot/fitImage\0"	\
 	"ec_disable_swsync=0\0"		\
-	"ec_image=ec-sulfur-rev3.RW.bin\0" \
-	"bootargs=root=/dev/mmcblk0p2 rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
+	"ec_image=lib/firmware/ni/ec-sulfur-rev3.RW.bin\0"	\
+	"bootargs=root=${mender_kernel_root} rw rootwait uio_pdrv_genirq.of_id=usrp-uio\0" \
 	"load_addr=0x2000000\0"		\
 	"fit_size=0x800000\0"           \
 	"fdt_high=0x20000000\0"         \
 	"initrd_high=0x20000000\0"      \
-	"sdboot=if test ${ec_disable_swsync} = 0; then \
+	"sdboot=run mender_setup; " \
+		"if test ${ec_disable_swsync} = 0; then "\
 			"echo Copying MCU FW from SD to RAM... && " \
-			"load mmc 0 ${load_addr} ${ec_image} && " \
+			"ext4load ${mender_uboot_root} ${load_addr} ${ec_image} && " \
 			"crosec swsync ${load_addr} ${filesize}; fi;" \
 		"echo Copying FIT from SD to RAM... && " \
-		"load mmc 0 ${load_addr} ${fit_image} && " \
-		"bootm ${load_addr}\0" \
+		"ext4load ${mender_uboot_root} ${load_addr} ${fit_image} && " \
+		"bootm ${load_addr}#conf@zynq-ni-${variant}.dtb\0" \
 	"jtagboot=echo TFTPing FIT to RAM... && " \
 		"tftpboot ${load_addr} ${fit_image} && " \
 		"bootm ${load_addr}\0" \
-- 
2.7.4

